### ISSUE TITLE
The lane_following can not operate normally in some conditions

### ISSUE THREAD

**Issue by shulanbushangshu** [autowarefoundation/autoware.universe#2633]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

When I use the map to do simulation about lane_following, there are three cases.

case1:
![Screenshot from 2023-01-10 13-14-00](https://user-images.githubusercontent.com/102840938/211480327-629571b3-30a6-49e2-997c-32be4046ee7f.png)

case2:
![Screenshot from 2023-01-10 13-16-53](https://user-images.githubusercontent.com/102840938/211480279-a470816e-d555-4ccc-b1d9-917f0b28332e.png)

case3:
![Screenshot from 2023-01-10 14-50-24](https://user-images.githubusercontent.com/102840938/211481330-d7d9c131-7f1e-452f-a8cf-22fea204f2c8.png)

In the case1 and case3,the mode can work normally.But in the case2,the mode is unable to operate normally.

### Expected behavior

In the conditions of case2,the lane_following is unable to operate normally.

### Actual behavior

In the conditions of case2,the lane_following is unable to operate normally.

### Steps to reproduce

1.Use the map of demo and start autoware planning simulator;
2. Set initial position in the lane of 34976
3. Set target in the lane of 35016

### Versions

1.OS: Ubuntun 20.04
2.ROS2: Galactic
3.Autoware: Autoware.universe:main

### Possible causes

In the path  based on sampling distance by parameters,there is the lane which  connects  to  the start lane.

**Comment by zulfaqar-azmi-t4** [comment]
@shulanbushangshu 
This PR https://github.com/autowarefoundation/autoware.universe/pull/2648 might interest you.

**Comment by mitsudome-r** [comment]
@shulanbushangshu Could you explain what you mean by "not operating normally"? Is it not engaging?

**Comment by shulanbushangshu** [comment]
@mitsudome-r No,it is engaged.I test the new code after the pr https://github.com/autowarefoundation/autoware.universe/pull/2648 and the result is that the path can not get the target lane as below shown:
![Screenshot from 2023-01-16 17-46-40](https://user-images.githubusercontent.com/102840938/213061056-a2769224-8764-4e90-91c4-c84b709631ee.png)

**Comment by purewater0901** [comment]
@shulanbushangshu Thank you for your reporting. I also encountered the same issue, but this problem is not related to [this PR](https://github.com/autowarefoundation/autoware.universe/pull/2648). I think the root cause of the problem comes from either Lanelet library or the route handler. The route handler(or Lanelet library) gives incorrect Lanelet sequences.

**Comment by purewater0901** [comment]
It seems [this funciton](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/route_handler/src/route_handler.cpp#L715-L729) chooses wrong lanelet in this case

**Comment by shulanbushangshu** [comment]
@purewater0901 Yes,this problem is not related to https://github.com/autowarefoundation/autoware.universe/pull/2648. The function of getNextLaneletWithinRoute() has no judgement when chose the next lane.
![Screenshot from 2023-01-18 12-46-21](https://user-images.githubusercontent.com/102840938/213086480-41e31431-28d4-4f7a-8816-3b6437ee6509.png)

**Comment by purewater0901** [comment]
@shulanbushangshu Yes, when you make a "half-looped" route, it won't work correctly. In order to fix this problem, we have to re-design the route handler.
